src/term/transform/scott_generation.rs

```diff
@@ -0,0 +1,69 @@
+use crate::term::{Book, Definition, Name, Rule, Term};
+
+impl Book {
+  pub fn generate_scott_adts(&mut self) {
+    for adt in self.adts.values() {
+      let curr_ctrs = adt.ctrs.clone();
+
+      for (ctr_name, arity) in &curr_ctrs {
+        let mut ctr_args = Vec::new();
+        let ctrs = curr_ctrs.keys().cloned().collect();
+        make_args(arity, &mut ctr_args);
```

Comment:
You could make this function return `ctr_args` instead for a cleaner interface.

`let ctr_args = make_args(arity);`